File: middlerApp.SharedModels/UpdateMiddlerRuleDto.cs
using System;
using System.Collections.Generic;
using middler.Common.SharedModels.Models;

namespace middler.Hosting.Models {
    public class UpdateMiddlerRuleDto {

        public string Name { get; set; }


        public List<string> Scheme { get; set; } = new List<string>();


        public string Hostname { get; set; }


        public string Path { get; set; }


        public List<string> HttpMethods { get; set; } = new List<string>();


        public List<MiddlerRulePermission> Permissions { get; set; } = new List<MiddlerRulePermission>();


        public List<MiddlerAction> Actions { get; set; } = new List<MiddlerAction>();

        public bool Enabled { get; set; }

        public decimal Order { get; set; }


    }
}

File: middlerApp.UI/projects/main/src/app/components/rules/models/update-middler-rule-dto.ts
import { MiddlerRulePermission } from './middler-rule-permission';
import { MiddlerAction } from './middler-action';

export class UpdateMiddlerRuleDto {
    Name: string;
    Scheme: Array<string>;
    Hostname: string;
    Path: string;
    HttpMethods: Array<string>;
    Permissions: Array<MiddlerRulePermission>;
    Actions: Array<MiddlerAction>;
}
